src/main/java/dev/bernasss12/bebooks/util/NBTUtils.java

```diff
@@ -45,51 +41,70 @@ public static ListTag sort(ListTag listTag, ModConfig.SortingSetting mode, boole
+        Stream<EnchantmentCompound> candidates = sorted(listTag, mode, true)
+            .filter(EnchantmentCompound::isRegistered);
 
+        Optional<EnchantmentCompound> priority;
         if (ModConfig.doCurseColorOverride && hasCurses(listTag)) {
-            return listTag.getCompound(listTag.size() - 1).getString("id");
+            priority = candidates.reduce((a, b) -> b); // last element
+        } else {
+            priority = candidates.findFirst();
         }
-        return sorted.getCompound(0).getString("id");
+
+        return priority.map(EnchantmentCompound::getId).orElse("");
     }
 
-    private static class EnchantmentCompound {
+    public static class EnchantmentCompound {
         @NotNull private final CompoundTag compound;
-        @NotNull private final String id;
-        @NotNull private final String translatedName;
-        private final int index;
-        private final boolean isCursed;
+        private final Enchantment enchantment;
+        private String id = null;
+        private String translatedName = null;
+        private int index = -1;
+        private boolean isCursed = false;
 
-        EnchantmentCompound(@NotNull Tag tag) {
+        public EnchantmentCompound(@NotNull Tag tag) {
             if (tag.getType() != 10) {
                 throw new AssertionError("tag is not a CompoundTag");
             }
 
             this.compound = (CompoundTag) tag;
-            this.id = compound.getString("id");
-            int lvl = compound.getShort("lvl");
 
-            Enchantment enchantment = Objects.requireNonNull(Registry.ENCHANTMENT.get(new Identifier(id)));
-            this.isCursed = enchantment.isCursed();
+            Identifier identifier = Identifier.tryParse(compound.getString("id"));
+            this.enchantment = Registry.ENCHANTMENT.get(identifier);
+
+            // Items can have non-registered enchantment tags on them
+            if (identifier == null || enchantment == null) {
+                // dummy comparison values
+                this.translatedName = "";
+                this.index = 0;
+                return;
+            }
+
+            this.isCursed = this.enchantment.isCursed();
+            this.id = identifier.toString(); // normalize e.g. "power" to "minecraft:power"
+        }
 
-            ModConfig.EnchantmentData enchantmentData = ModConfig.enchantmentDataMap.get(id);
+        private void lazyInit() {
+            ModConfig.EnchantmentData enchantmentData = ModConfig.enchantmentDataMap.get(this.id);
             if (enchantmentData != null) {
-                this.translatedName = enchantmentData.translatedName;
+                this.translatedName = enchantmentData.getTranslatedName();
                 this.index = enchantmentData.orderIndex;
             } else {
-                this.translatedName = enchantment.getName(lvl).asString();
+                this.translatedName = I18n.translate(this.enchantment.getTranslationKey());
```

Comment:
`enchantment.getName(lvl).asString()` would return an empty string, which is why sorting was broken, `getString()` would have been right.
Since this is used for sorting and the config screen, I left out the levels.